### PR TITLE
New validate sub-plugin "config"

### PR DESCRIPTION
**Pull request by Qalthos**
##### SUMMARY

Implement https://github.com/ansible-collections/ansible.network/issues/15 as a validate sub-plugin.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

validate

### CODE REVIEW

**Comment by Qalthos** [comment]
recheck

**Review by ganeshrn — COMMENTED** [review]
Can you please update the example ti use this sub-plugin https://github.com/ansible-collections/ansible.utils/blob/main/plugins/modules/validate.py#L51

**Review by ganeshrn — COMMENTED** [review]
Also please add integration tests https://github.com/ansible-collections/ansible.utils/tree/main/tests/integration/targets/utils_validate

**Review by ganeshrn — APPROVED** [review]
Looks good. Thank you!

File: plugins/modules/validate.py [hunk 1]
```diff
@@ -61,6 +61,12 @@
     engine: ansible.utils.jsonschema
   vars:
     ansible_jsonschema_draft: draft7
+
+- name: validate configuration with config plugin (see config plugin for criteria examples)
```
Comment (NilashishC) [inline plugins/modules/validate.py:65]: Do we also want to add a few examples rules in the docs?

**Comment by Qalthos** [comment]
recheck

File: plugins/modules/validate.py (hunk 1 above)
Comment (Qalthos) [reply plugins/modules/validate.py:65]: I'm not sure this file is the place to do that? This example has as much detail as the jsonschema one above, and there are example rules in the sub-plugin, we just need to find a good way to expose them.

**Pull request merged**